lms/djangoapps/courseware/models.py

```diff
@@ -260,3 +260,20 @@ class Meta:
 
     def __unicode__(self):
         return "[OCGLog] %s: %s" % (self.course_id.to_deprecated_string(), self.created)  # pylint: disable=no-member
+
+
+class StudentFieldOverride(models.Model):
+    """
+    Holds the value of a specific field overriden for a student.  This is used
+    by the code in the `courseware.student_field_overrides` module to provide
+    overrides of xblock fields on a per user basis.
+    """
+    course_id = CourseKeyField(max_length=255, db_index=True)
+    location = LocationKeyField(max_length=255, db_index=True)
+    student = models.ForeignKey(User, db_index=True)
+
+    class Meta:   # pylint: disable=missing-docstring
+        unique_together = (('course_id', 'location', 'student'),)
+
+    field = models.CharField(max_length=255)
+    value = models.TextField(default='null')
```

Comment:
@ormsbee 1) Doesn't make much sense as written.  I'm not sure why you're distinguishing between (course_id, location) and (course_id, student) queries when the index is on all 3.  

2) Sure.  (One of my colleagues can do this.)

3) I don't really understand what you're getting at.  If there is something concrete we can do to address your concern, let us know.

4) The rationale for leaving out field is that the number of fields in a particular block for a particular student is going to be quite small, so there's not much point in indexing further down than that.  In the POCs/CCX branch, this has been rewritten such that all fields for a (course_id, location, student) are read at once and cached to make things a little more efficient, which further reduces the need for an index, since it would never be used.

5) I wasn't aware we were keeping a history of student state.  Is that a requirement for this?